### PR TITLE
DGtal on MS Windows with QGLViewer activated

### PR DESCRIPTION
**Pull request by kerautret**
This PR focus on remaining errors when QGLviewer is installed and activated.
It is  not yet finished:
- [X] Install QGLViewer with appveyor
- [x] main DGtal with QGLviewer activated
- [x] Check Examples section : 
   - [x] fix gl errors (add #include<windows.h>)
   - [x] fix CATCH error (from exampleCatch.h)  
- [x] Check Tests section

### CODE REVIEW

**Comment by dcoeurjo** [comment]
Excellent..

**Comment by dcoeurjo** [comment]
the problem on appveyor is that we usually have timeout while compiling the tests...

**Comment by kerautret** [comment]
yes....  I got it on Windows 10: (win32)
capture d ecran 2016-01-14 a 14 05 26

**Comment by kerautret** [comment]
for the errors on catch, I have the same errors on real device ;(

**Comment by kerautret** [comment]
@dcoeurjo  If you want here is error:
C:\projets\DGtal\examples\doc-examples\exampleCatch.cpp(52): error C2448: 'TEST_CASE' : l'initialiseur de style fonction semble être une définition de fonction [C:\projets\DGtal\build\examples\doc-examples\exampleCatch.vcxproj]
C:\projets\DGtal\examples\doc-examples\exampleCatch.cpp(97): error C2065: 'T' : identificateur non déclaré [C:\projets\DGtal\build\examples\doc-examples\exampleCatch.vcxproj]
C:\projets\DGtal\examples\doc-examples\exampleCatch.cpp(98): error C2275: 'std::list<int,std::allocator<_Ty>>' : utilisation non conforme de ce type comme expression [C:\projets\DGtal\build\examples\doc-examples\exampleCatch.vcxproj]

If you have an idea...

**Comment by kerautret** [comment]
I fix it, it was simply the symbolic link which was not taken into account from the git of windows...
I replace it by a copy of the source file.

**Comment by kerautret** [comment]
all done !  thats compile and run on windows...

**Comment by dcoeurjo** [comment]
Nice snapshot, congrats !

**Comment by kerautret** [comment]
thanks,;)
There are some bug on reading some vol/pgm3d I particular by using some tools of DGtalTools or in other example (perhaps due to type when reading input file).

File: tests/base/testSetFunctions.cpp [hunk 1]
```diff
@@ -40,7 +40,11 @@
 
 using namespace DGtal;
 using namespace DGtal::functions;
-using namespace DGtal::functions::setops;
+//using  DGtal::functions::setops::operator|;
```
Comment (dcoeurjo) [inline tests/base/testSetFunctions.cpp]: what's wrong with "|" operator ?

**Comment by dcoeurjo** [comment]
thanks for the PR.Merging
(but please tell me for "|" operator)

**Comment by dcoeurjo** [comment]
Can I reactivate appevayor on DGtal master ?

**Comment by dcoeurjo** [comment]
(if the build is too long and timeout, at least it would be great to have the example compiled).

**Comment by kerautret** [comment]
For the "|" operator it looks strange, but when using we have:
  testSetFunctions.cpp
C:\Qt\5.4\msvc2013_opengl\include\QtCore/qmetatype.h(907): error C2057: expression constante attendue
  C:\Qt\5.4\msvc2013_opengl\include\QtCore/qmetatype.h(917): note: voir la référence à l'instanciation de la classe modèle 'QtMetaTypePrivate::CapabilitiesImpl<QList<QVariant>,std::random_access_iterator_tag>' en cours de compilation
  C:\Qt\5.4\msvc2013_opengl\include\QtCore/qmetatype.h(988): note: voir la référence à l'instanciation de la classe modèle 'QtMetaTypePrivate::ContainerAPI<T>' en cours de compilation
          with
          [
              T=QVariantList
          ]
  C:\Qt\5.4\msvc2013_opengl\include\QtCore/qvariant.h(714): note: voir la référence à l'instanciation de la fonction modèle 'QtMetaTypePrivate::QSequentialIterableImpl::QSequentialIterableImpl<QVariantList>(const T *)' en cours de compilation
          with
          [
              T=QVariantList
          ]

**Comment by kerautret** [comment]
I add an issue on that.
Yes of course for appveyor, it should work you can just set the example.

**Comment by JacquesOlivierLachaud** [comment]
Le 14/01/2016 14:07, Bertrand Kerautret a écrit :
> yes.... I got it on Windows 10: (win32)
> capture d ecran 2016-01-14 a 14 05 26
> <https://cloud.githubusercontent.com/assets/772865/12325088/094f2086-bac8-11e5-9905-ebbea7b3079f.png>
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DGtal-team/DGtal/pull/1106#issuecomment-171638964>.
> 
Congrats Bertrand !

**Pull request merged**